src/App.jsx

```diff
@@ -0,0 +1,400 @@
+        ]
+      });
+    } catch (err) {
+      console.log(err);
+      console.log('No Pledge Found!');
+    }
+
+    setPledge(pledge);
+  };
+
+  const pawn = async () => {
+    try {
+      const transactionId = await fcl.mutate({
+        cadence: `${pawnNFTs}`,
+        args: (arg, t) => [
+          arg(selectedCollection, t.String),
+          arg(selectedNFTs, t.Array(t.UInt64))
+        ],
+        proposer: fcl.currentUser,
+        payer: fcl.currentUser,
+        limit: 1000,
+      });
+
+      console.log(
+        'Pawning NFTs now with transaction ID',
+        transactionId
+      );
+
+      const transaction = await fcl.tx(transactionId).onceSealed();
+
+      console.log(
+        'Testnet explorer link:',
+        `https://testnet.flowscan.org/transaction/${transactionId}`
+      );
+
+      console.log(transaction);
+      alert('NFTs sold successfully!');
+    } catch (error) {
+      console.log(error);
+      alert(
+        'Error selling NFT, please check the console for error details!'
+      );
+    }
+  };
+
+  const redeem = async () => {
+    try {
+      const transactionId = await fcl.mutate({
+        cadence: `${redeemPledge}`,
+        args: (arg, t) => [
+          arg(pledge.pawns[0].collectionIdentifier, t.String)
+        ],
+        proposer: fcl.currentUser,
+        payer: fcl.currentUser,
+        limit: 999,
+      });
+
+      console.log(
+        'Redeem an NFT now with transaction ID',
+        transactionId
+      );
+
+      const transaction = await fcl
+        .tx(transactionId)
+        .onceSealed();
+
+      console.log(
+        'Testnet explorer link:',
+        `https://testnet.flowscan.org/transaction/${transactionId}`
+      );
+
+      console.log(transaction);
+      alert('NFT redeem successfully!');
+    } catch (error) {
+      console.log(error);
+      alert(
+        'Error redeeming NFT, please check the console for error details!'
+      );
+    }
+  };
+
+  const RenderLogout = () => {
+    if (user && user.addr) {
+      return (
+        <div className="logout-container">
+          <button
+            className="cta-button logout-btn"
+            onClick={() => logOut()}
+          >
+            ❎ {'  '}
+            {user.addr.substring(0, 6)}...
+            {user.addr.substring(user.addr.length - 4)}
+          </button>
+        </div>
+      );
+    }
+
+    return undefined;
+  };
+
+  return (
+    <div className="App">
+      <RenderLogout />
+      <div className="container">
+        <div className="header-container">
+          <div className="logo-container">
+            <img src="./logo.png" className="flow-logo" alt="flow logo" />
+            <p className="header">
+              ✨ Best NFT Pawnshop ever on Flow ✨
+            </p>
+          </div>
+
+          <p className="sub-text">
+            Harvesting your NFT losses in just a few clicks.
+          </p>
+        </div>
+
+        {user && user.addr ? (
+          <React.Fragment>
```

Comment:
I am thinking we need to create Separate this one from App.tsx. 
1. We need to create separate component like WalletComponent.tsx where we will put alll wallet related thing.
2. ANother is login and logout related.

Because we need to clean App.tsx as it entry point of application. so keep it simple and breakout possible things.